sonar-update-center-common/src/main/java/org/sonar/updatecenter/common/Component.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * SonarSource :: Update Center :: Common
+ * Copyright (C) 2010-2019 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.updatecenter.common;
+
+import java.util.List;
+
+public abstract class Component extends Artifact {
+
+  protected String name;
+  protected String description;
+  protected String homepageUrl;
+  protected String license;
+  protected String organization;
+  protected String organizationUrl;
+  protected String termsConditionsUrl;
+  protected String category;
+  protected String issueTrackerUrl;
+  protected String sourcesUrl;
+  protected List<String> developers;
+
+  protected Component(String key) {
+    super(key);
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  public Component setName(String name) {
+    this.name = name;
+    return this;
+  }
+
+  public String getDescription() {
+    return description;
+  }
+
+  public Component setDescription(String description) {
+    this.description = description;
+    return this;
+  }
+
+  public String getHomepageUrl() {
+    return homepageUrl;
+  }
+
+  public Component setHomepageUrl(String s) {
+    this.homepageUrl = s;
```

Comment:
NP: I know this is copy-pasted from old code, but for all other methods, we type out the parameter fully (`description`, `license`, `url`, `name`, ...) For consistency, I would name this parameter at least `url` as well.